routes/projects/files/rename.js

```diff
@@ -0,0 +1,23 @@
+/*
+ * PUT /projects/:username/:projectName/files/:pathToFile/rename
+ * The :pathToFile parameter must be URI encoded
+ * Renames the specified file with the new name given in the message body
+ * {
+ *   name: <string> required
+ * }
+*/
+
+var renameSchema = require('../../../schemas.js').rename;
+
+module.exports = function(server) {
+  server.route({
+    method: 'PUT',
+    path: '/projects/{username}/{projectName}/files/{path}/rename',
```

Comment:
So we'll have to do some work to maintain consistency across the backend storage view of a file vs. the filesystem, where the filesystem will only ever let us rename the leaf node.  It might be useful for us to disallow renaming `/public/styles/custom.css` to `/public/custom.css`, since it will lead to filesystem data loss.